pyasdf/fits_embed.py

```diff
@@ -0,0 +1,156 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+# -*- coding: utf-8 -*-
+
+from __future__ import absolute_import, division, unicode_literals, print_function
+
+"""
+Utilities for embedded ADSF files in FITS.
+"""
+import io
+import re
+
+import numpy as np
+
+from astropy.extern import six
+from astropy.io import fits
+
+from . import asdf
+from . import block
+from . import util
+
+
+ASDF_EXTENSION_NAME = 'ASDF'
+
+FITS_SOURCE_PREFIX = 'fits:'
+
+
+class _FitsBlock(object):
```

Comment:
This is to support storing an ASDF file inside of a FITS HDU, and that ASDF content refers to data in the regular FITS HDUs in the same file.

So a FITS block references a FITS HDU by an `EXTNAME,EXTVER` pair.  It could be an offset from the start of the file, I suppose, but this would allow the FITS headers in the file to be edited by eg. `astropy.io.fits` without breaking the ASDF portion.